### PR TITLE
fed_sender: Console.ReadKey() => lock with AutoResetEvent

### PR DESCRIPTION
**Pull request by krombel**

### CODE REVIEW

**Comment by Half-Shot** [comment]
LGTM, removes our need for an interactive term. What's your feelings on this @turt2live ?

File: Matrix.SynapseInterop.Worker.FederationSender/Program.cs [hunk 1]
```diff
@@ -32,9 +35,17 @@ private static void Main(string[] args)
...
             new FederationSender(_config).Start().Wait();
 
-            Console.ReadKey(true);
+            _closing.WaitOne();
+        }
+
+        protected static void OnExit(object sender, ConsoleCancelEventArgs args)
+        {
+                Console.WriteLine("Exit");
+                _closing.Set();
```
Comment (turt2live) [inline Matrix.SynapseInterop.Worker.FederationSender/Program.cs]: The indentation on these 2 lines is wrong

**Review by turt2live — APPROVED** [review]
otherwise lgtm

File: Matrix.SynapseInterop.Worker.FederationSender/Program.cs (hunk 1 above)
Comment (krombel) [reply Matrix.SynapseInterop.Worker.FederationSender/Program.cs]: done

**Comment by Half-Shot** [comment]
Thaaaaaaaaaaaaaaaanks :)

**Pull request merged**